### PR TITLE
Prefix local storage access with user email address

### LINKED ISSUE #4190: Ensure all localstorage keys depend on user name
For example, simply prefixing all localstorage keys with the user email address would already solve that layouts are shared across accounts when the same computer is used.

### PR DESCRIPTION
**Pull request by philippotto**
According to `main.js`, the entire rendering of views is blocked by loading the active user. So, simply depending on the store synchronously should work pretty well. For the layouts, my testing showed that everything works as expected, except for logging in on a tracing page without reloading. One could expect that the layout is changed automatically (to the last layout used by the logged in user). However, the layout code itself is quite brittle which is why I didn't want to touch that. Also, the impact isn't really big. What happens now is that the layout stays the same, but after reloading the last active layout (by the user) is loaded again.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- change the layout while being logged in (and off)
- log in / out and check that the layouts are persisted 

### Issues:
- fixes #4190 

------
- [x] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)
- [X] Ready for review

### CODE REVIEW

File: frontend/javascripts/oxalis/view/layouting/layout_persistence.js [hunk 1]
```diff
@@ -80,16 +82,26 @@ function readStoredLayoutConfigs() {
   return defaultLayoutConfig;
 }
 
-Store.dispatch(setStoredLayoutsAction(readStoredLayoutConfigs()));
+listenToStoreProperty(
+  storeState => storeState.activeUser,
+  () => {
+    console.log("Dispatching setStoredLayoutsAction");
```
Comment (daniel-wer) [inline frontend/javascripts/oxalis/view/layouting/layout_persistence.js]:
```suggestion
```

File: frontend/javascripts/libs/user_local_storage.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+// @flow
+
+import Store from "oxalis/store";
+
+function prefixKey(key) {
+  const { activeUser } = Store.getState();
+  console.log("activeUser", activeUser);
```
Comment (daniel-wer) [inline frontend/javascripts/libs/user_local_storage.js]:
```suggestion
```

File: frontend/javascripts/libs/user_local_storage.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  const prefix = !activeUser ? "Anonymous" : activeUser.email;
+  return `${prefix}-${key}`;
+}
+
+const UserLocalStorage = {
+  getItem(key: string): ?string {
+    console.log("getting", key);
```
Comment (daniel-wer) [inline frontend/javascripts/libs/user_local_storage.js]:
```suggestion
```

File: frontend/javascripts/libs/user_local_storage.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    return localStorage.getItem(prefixKey(key));
+  },
+
+  setItem(key: string, value: string): void {
+    console.log("setting", key);
```
Comment (daniel-wer) [inline frontend/javascripts/libs/user_local_storage.js]:
```suggestion
```

File: frontend/javascripts/libs/user_local_storage.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    return localStorage.setItem(prefixKey(key), value);
+  },
+
+  clear(): void {
+    localStorage.clear();
+  },
+
+  removeItem(key: string): void {
+    return localStorage.removeItem(key);
```
Comment (daniel-wer) [inline frontend/javascripts/libs/user_local_storage.js]: Shouldn't this also use `prefixKey(key)`?

Comment (philippotto) [reply frontend/javascripts/libs/user_local_storage.js]: Whoops, well spotted! :+1:

**Comment by philippotto** [comment]
Thanks for your review and the commit suggestions :+1: Should be ready now :)

**Pull request merged**